src/main/java/io/supertokens/webserver/api/multitenancy/thirdparty/CreateOrUpdateThirdPartyConfigAPI.java

```diff
@@ -112,9 +113,12 @@ protected void doPut(HttpServletRequest req, HttpServletResponse resp) throws IO
                     tenantConfig.emailPasswordConfig,
                     new ThirdPartyConfig(
                             tenantConfig.thirdPartyConfig.enabled,
+                            getVersionFromRequest(req).lesserThan(SemVer.v5_1) ?
```

Comment:
why checking version here? The value of useThirdPartyProvidersFromStaticConfigIfEmpty should be already set correctly based on when it was created.

Reply:
yes, but when add/update/delete thirdParty config, we should set it to false for CDI >= 5.1. otherwise, preserve the older state